src/Databases/DataLake/ICatalog.h

```diff
@@ -159,9 +159,11 @@ class ICatalog
     /// E.g. one of S3, Azure, Local, HDFS.
     virtual std::optional<StorageType> getStorageType() const = 0;
 
-    virtual void createTable(const String & namespace_name, const String & table_name, const String & new_metadata_path) const;
+    virtual void createTable(const String & namespace_name, const String & table_name, const String & new_metadata_path, Poco::JSON::Object::Ptr metadata_content) const;
 
-    virtual void updateMetadata(const String & namespace_name, const String & table_name, const String & new_metadata_path) const;
+    virtual bool updateMetadata(const String & namespace_name, const String & table_name, const String & new_metadata_path, Poco::JSON::Object::Ptr new_snapshot) const;
+
+    virtual bool isTransactional() const { return false; }
```

Comment:
Need to add a small comment  above this method explaining what this method should mean (as you can see almost all methods in this class have comments, it is a good thing to do)